src/Services/SearchService.php

```diff
@@ -288,15 +293,15 @@ private function getProductDetailUrl(int $productId)
         /** @var ItemSearchService $itemSearchService */
         $itemSearchService = $this->getItemSearchService();
 
-        $externalSearchFactories[$productId] = $this->getSearchFactory($productId);
-
-        $result = $itemSearchService->getResults($externalSearchFactories);
+        $result = $itemSearchService->getResult(
+            $this->getSearchFactory([$productId])
+        );
 
-        if (empty($result[$productId]['documents'])) {
+        if (!$result['success'] && empty($result['documents'][0])) {
```

Comment:
Shouldn't this be an `||`? You do not want to continue if you had an item but no success and vise-versa.